### PR TITLE
Add lance.util.duckdb to help install the extension transparently

### PR DESCRIPTION
**Pull request by changhiskhan**
It's too complicated to make users go through the installation process themselves. Instead let's create a small python utility function. In a python-centric workflow, this will be the easiest method. Alternatives can also be considered like providing a torch-like website or a command line utility that the user needs to download.

### CODE REVIEW

File: python/lance/util/duckdb_ext.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import duckdb
+import os
+import pathlib
+import platform
+import urllib.request
+import zipfile
+
+
+def install_duckdb_extension(version='latest', unsigned=True):
```
Comment (eddyxu) [inline python/lance/util/duckdb_ext.py:9]: we dont need `unsigned`, which is always true if install `lance`?

File: python/lance/util/duckdb_ext.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    uri_root = 'https://eto-public.s3.us-west-2.amazonaws.com/'
+    uri = os.path.join(uri_root, 'artifacts', 'lance', 'lance_duckdb',
+                       version, zip_name)
+    return uri
+
+
+def _get_device():
+    import torch
+    if torch.cuda.is_available():
+        return f"cu{torch.version.cuda.replace('.', '')}"
```
Comment (eddyxu) [inline python/lance/util/duckdb_ext.py:44]: Pytorch usually is built with a few cuda versions. do we want to support them all?

File: python/lance/util/duckdb_ext.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import duckdb
```
Comment (eddyxu) [inline python/lance/util/duckdb_ext.py:1]: Please add License.

File: python/lance/util/duckdb_ext.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    import torch
+    if torch.cuda.is_available():
+        return f"cu{torch.version.cuda.replace('.', '')}"
+    else:
+        return 'cpu'
+
+
+def _get_latest_version(org='lance', ext='lance_duckdb'):
+    uri_root = 'https://eto-public.s3.us-west-2.amazonaws.com/'
+    uri = os.path.join(uri_root, 'artifacts', org, ext, 'latest')
```
Comment (eddyxu) [inline python/lance/util/duckdb_ext.py:51]: should we check duckdb version and torch version?

i.e., can duckdb `0.5.0` loads the extension built against `0.5.1`

Comment (changhiskhan) [reply python/lance/util/duckdb_ext.py:51]: good question i can try it first. If it's at least binary compatible across minor versions then we can build against 0.X.0 and just check the version

File: python/lance/util/duckdb_ext.py (hunk 2 above)
Comment (changhiskhan) [reply python/lance/util/duckdb_ext.py:44]: i think that's too much infra for us to take on right now unless there's a way we can switch between different cuda versions in a docker container or smth?

Comment (changhiskhan) [reply python/lance/util/duckdb_ext.py:44]: i think that's too much infra for us to take on right now unless there's a way we can switch between different cuda versions in a docker container or smth?

File: python/lance/util/duckdb_ext.py (hunk 3 above)
Comment (changhiskhan) [reply python/lance/util/duckdb_ext.py:14]: done

File: python/lance/util/duckdb_ext.py (hunk 1 above)
Comment (changhiskhan) [reply python/lance/util/duckdb_ext.py]: done

File: python/lance/util/duckdb_ext.py (hunk 4 above)
Comment (changhiskhan) [reply python/lance/util/duckdb_ext.py:62]: ok so unfortunately the answer is no:

```python
In [5]: con.load_extension("lance")
---------------------------------------------------------------------------
InvalidInputException                     Traceback (most recent call last)
Cell In [5], line 1
----> 1 con.load_extension("lance")

InvalidInputException: Invalid Input Error: Extension "/Users/changshe/.duckdb/extensions/109f932c4/osx_arm64/lance.duckdb_extension" version (v0.5.1) does not match DuckDB version (0.5.0)
```

**Review by eddyxu — APPROVED** [review]
LGTM

**Pull request merged**